All candidates page

JamesHuf opened the pull request:
Resolves #3688 and should resolve #3687.

Updates the existing pool candidate table to use backend pagination and contain different fields. Also creates a simple csv for pool candidate data.

Remaining:
- ~Breadcrumbs to page~
- ~Subtitle for pool~
- Download pdf behavior for pool candidates
- ~Review what fields should be included in pdf and csv~

vd1992 commented on api/graphql/schema.graphql:
```diff
@@ -738,7 +742,8 @@ type Query {
     pools: [Pool]! @all
     poolCandidate(id: ID! @eq): PoolCandidate @find @guard @can(ability: "view", query: true)
-    poolCandidates: [PoolCandidate]! @all(scopes: ["expiryFilter"]) @guard @can(ability: "viewAny")
+    poolCandidates(includeIds: [ID] @in(key: "id")): [PoolCandidate]! @all(scopes: ["expiryFilter"]) @guard @can(ability: "viewAny")
```
something that comes to mind, we will probably want to remove the expiry scope from this as there will be toggled filters that allow viewing of expired candidates say for printing purposes
if the scope remains you get the oddity of being able to select the expired candidate due to the poolCandidatesPaginated query but unable to print or do anything due to the poolCandidates query auto-removing expired

JamesHuf replied:
I've now removed the scope from the query.

esizer commented on frontend/common/src/components/PageHeader/PageHeader.tsx:
```diff
@@ -3,11 +3,42 @@ import Heading from "../Heading";
 
 export interface PageHeaderProps {
   icon?: React.FC<{ className: string }>;
+  subtitle?: string;
```
In order to allow for `react-intl` messages this should be the type `React.ReactNode`

```suggestion
  subtitle?: React.ReactNode;
```

JamesHuf replied:
Changed with [Changed subtitle in PageHeader to be ReactNode](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3956/commits/7c43873984aef83e6a92a796b6b62b574513c43c)

esizer commented on frontend/admin/src/js/components/poolCandidate/PoolCandidatesTable.tsx:
```diff
@@ -1,217 +1,478 @@
+      return (
+        <span data-h2-color="base(dt-primary)" data-h2-font-weight="base(700)">
+          {intl.formatMessage({
+            defaultMessage: "Priority Entitlement",
+            id: "j1p7LR",
+            description: "Priority text for users with priority entitlement",
+          })}
+        </span>
+      );
+    case 20:
+      return (
+        <span data-h2-color="base(dt-primary)" data-h2-font-weight="base(700)">
+          {intl.formatMessage({
+            defaultMessage: "Veteran",
+            id: "oU8C65",
+            description: "Priority text for veterans",
+          })}
+        </span>
+      );
+    case 30:
+      return (
+        <span>
+          {intl.formatMessage({
+            defaultMessage: "Citizen or Resident",
+            id: "oMyc4e",
+            description: "Priority text for citizens of canada",
+          })}
+        </span>
+      );
+    default:
+      return (
+        <span>
+          {intl.formatMessage({
+            defaultMessage: "Work Visa",
+            id: "EimWiB",
+            description: "Priority text for users with work visas",
+          })}
+        </span>
+      );
+  }
```
Hmm, if we plan on using this elsewhere could we move this to `localizedConstants.tsx` with something like:

```tsx
export const poolCandidatePriorities = defineMessages({
  [10]: {
    defaultMessage: "Priority Entitlement",
    ...
  },
  ...
});

```

JamesHuf replied:
Added with [Moved status text to localized constants and changed 'Work Visa' to 'Other'](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3956/commits/4bddfcc5e268de11913d07daf3ec5aca523ed483)

esizer reviewed (changes requested):
Wow, this is looking really great! I just left two requests. Also, are we moving the print to a new issue to reduce scope or is it something you are still working on?

vd1992 commented:
it is a bit odd to use "Work Visa" when we are using "Other" elsewhere
I went with Other based off this #3749 
![image](https://user-images.githubusercontent.com/40485260/190691500-29290390-2f3f-4c96-b17f-1d5e8929b315.png)

JamesHuf commented:
> it is a bit odd to use "Work Visa" when we are using "Other" elsewhere I went with Other based off this #3749 ![image](https://user-images.githubusercontent.com/40485260/190691500-29290390-2f3f-4c96-b17f-1d5e8929b315.png)

Updated with [Moved status text to localized constants and changed 'Work Visa' to 'Other'](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3956/commits/4bddfcc5e268de11913d07daf3ec5aca523ed483)

JamesHuf commented:
Also I have added print functionality now

esizer reviewed (approved):
Awesome! Looking great 👍
